components/server/ee/src/user/user-deletion-service.ts

```diff
@@ -10,38 +10,58 @@ import { SubscriptionService } from "@gitpod/gitpod-payment-endpoint/lib/account
 
     async deleteUser(id: string): Promise<void> {
         const user = await this.db.findUserById(id);
         if (!user) {
             throw new Error(`No user with id ${id} found!`);
         }
 
+        const errors = [];
         if (this.env.enablePayment) {
-            const now = new Date().toISOString();
-            const subscriptions = await this.subscriptionService.getNotYetCancelledSubscriptions(user, now);
+            const now = new Date();
+            const subscriptions = await this.subscriptionService.getNotYetCancelledSubscriptions(user, now.toISOString());
             for (const subscription of subscriptions) {
-                const planId = subscription.planId!;
-                if (Plans.isFreeNonTransientPlan(planId)) {
-                    // only delete those plans that are persisted in the DB
-                    await this.subscriptionService.unsubscribe(user.id, now, planId);
-                } else if (Plans.isFreePlan(planId)) {
-                    // we do not care about transient plans
-                    continue;
-                } else {
-                    // cancel Chargebee subscriptions
-                    const subscriptionId = subscription.uid;
-                    const chargebeeSubscriptionId = subscription.paymentReference!;
-                    await this.chargebeeService.cancelSubscription(chargebeeSubscriptionId, { userId: user.id }, { subscriptionId, chargebeeSubscriptionId });
+                try {
+                    const planId = subscription.planId!;
+                    const paymentReference = subscription.paymentReference;
+                    if (Plans.isFreeNonTransientPlan(planId)) {
+                        // only delete those plans that are persisted in the DB
+                        await this.subscriptionService.unsubscribe(user.id, now.toISOString(), planId);
+                    } else if (Plans.isFreePlan(planId)) {
+                        // we do not care about transient plans
+                        continue;
+                    }
+                    if (!paymentReference) {
+                        const teamSlots = await this.teamSubscriptionService.findTeamSubscriptionSlotsByAssignee(id);
+                        teamSlots.forEach(async ts => await this.teamSubscriptionService.deactivateSlot(ts.teamSubscriptionId, ts.id, now))
+                    } else if (paymentReference.startsWith("github:")) {
+                        throw new Error("Cannot delete user subscription from GitHub")
+                    } else {
+                        // cancel Chargebee subscriptions
+                        const subscriptionId = subscription.uid;
+                        const chargebeeSubscriptionId = subscription.paymentReference!;
+                        await this.chargebeeService.cancelSubscription(chargebeeSubscriptionId, { userId: user.id }, { subscriptionId, chargebeeSubscriptionId });
+                    }
```

Comment:
Is it on purpose that this block is not in an else block anymore? Is it because team subscriptions have a free plan but no `paymentReference`? I would like to have a short comment here that explains this.

```suggestion
                    } else {
                        if (!paymentReference) {
                            const teamSlots = await this.teamSubscriptionService.findTeamSubscriptionSlotsByAssignee(id);
                            teamSlots.forEach(async ts => await this.teamSubscriptionService.deactivateSlot(ts.teamSubscriptionId, ts.id, now))
                        } else if (paymentReference.startsWith("github:")) {
                            throw new Error("Cannot delete user subscription from GitHub")
                        } else {
                            // cancel Chargebee subscriptions
                            const subscriptionId = subscription.uid;
                            const chargebeeSubscriptionId = subscription.paymentReference!;
                            await this.chargebeeService.cancelSubscription(chargebeeSubscriptionId, { userId: user.id }, { subscriptionId, chargebeeSubscriptionId });
                        }
                    }
```